extensions/guacamole-auth-ldap/src/main/java/org/glyptodon/guacamole/auth/ldap/connection/ConnectionService.java

```diff
@@ -109,11 +109,38 @@
             // possibly be null
             assert(userDN != null);
 
-            // Find all Guacamole connections for the given user
+            // If group base DN is specified search for user groups
+            String groupBaseDN = confService.getGroupBaseDN();
+            // Create a separate filter for user groups
+            StringBuilder userRoleGroupFilter = new StringBuilder();
+            if (groupBaseDN != null) {
+
+	            // Get all groups the user is a member of, excluding guacConfigGroups
+	            LDAPSearchResults userRoleGroupResults = ldapConnection.search(
+	            		groupBaseDN,
+	                    LDAPConnection.SCOPE_SUB,
+	                    "(&(!(objectClass=guacConfigGroup))(member=" + escapingService.escapeLDAPSearchFilter(userDN) + "))",
+	                    null,
+	                    false
+	            );
+
+	            // Create filter with seeAlso groups, the seeAlso attribute on
+	            // the guacConfigGroup will refer to other groups, any members
+	            // in these groups have access to the connection also
+	        	while (userRoleGroupResults.hasMore()) {
+	                LDAPEntry entry = userRoleGroupResults.next();
+	                userRoleGroupFilter.append("(seeAlso=").append(escapingService.escapeLDAPSearchFilter(entry.getDN())).append(")");
+	        	}
+            }
+
+        	// Find all Guacamole connections for the given user by
+        	// looking for direct membership in the guacConfigGroup
+        	// and possibly any groups the user is a member of that are
+            // referred to in the seeAlso attribute of the guacConfigGroup
             LDAPSearchResults results = ldapConnection.search(
                 configurationBaseDN,
                 LDAPConnection.SCOPE_SUB,
-                "(&(objectClass=guacConfigGroup)(member=" + escapingService.escapeLDAPSearchFilter(userDN) + "))",
+                "(&(objectClass=guacConfigGroup)(|(member=" + escapingService.escapeLDAPSearchFilter(userDN) + ")" + userRoleGroupFilter.toString() + "))",
```

Comment:
Readability is definitely extremely important. Perhaps there is a way that is clean *and* doesn't concatenate a `StringBuilder`?

Off the top of my head, moving the filter generation to its own private function might help smooth things, but I'm sure there are any number of ways to achieve this.